### PR TITLE
[ioredis] Fix mset and msetnx allowing object and string array

### PR DESCRIPTION
**Pull request by Sorosliu1029**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33190>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/ioredis/index.d.ts [hunk 1]
```diff
@@ -272,7 +272,7 @@ declare namespace IORedis {
         hgetBuffer(key: KeyType, field: string, callback: (err: Error, res: Buffer) => void): void;
         hgetBuffer(key: KeyType, field: string): Promise<Buffer>;
 
-        hmset(key: KeyType, field: string, value: any, ...args: string[]): Promise<0 | 1>;
+        hmset(key: KeyType, ...args: string[]): Promise<0 | 1>;
```
Comment (plantain-00) [inline types/ioredis/index.d.ts]: Why change this?

File: types/ioredis/index.d.ts [hunk 2]
```diff
@@ -673,7 +677,7 @@ declare namespace IORedis {
         hget(key: KeyType, field: string, callback?: (err: Error, res: string | string) => void): Pipeline;
         hgetBuffer(key: KeyType, field: string, callback?: (err: Error, res: Buffer) => void): Pipeline;
 
-        hmset(key: KeyType, field: string, value: any, ...args: string[]): Pipeline;
+        hmset(key: KeyType, ...args: string[]): Pipeline;
```
Comment (plantain-00) [inline types/ioredis/index.d.ts]: Why change this?

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/ioredis/index.d.ts (hunk 1 above)
Comment (Sorosliu1029) [reply types/ioredis/index.d.ts]: if `hmset` multi field-value pairs, there is no need to `hmset(key, field1, value1, ...[field2, value2])`, just `hmset(key, ...[field1, value1, field2, value2])` should works.

File: types/ioredis/index.d.ts (hunk 2 above)
Comment (Sorosliu1029) [reply types/ioredis/index.d.ts]: same as https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33190#discussion_r258020914

**Review by plantain-00 — APPROVED** [review]

File: types/ioredis/index.d.ts (hunk 1 above)
Comment (amcasey) [reply types/ioredis/index.d.ts]: Looks like it `value1` used to be `any` but now it's `string`.  Was that change intentional?

Comment (Sorosliu1029) [reply types/ioredis/index.d.ts]: Oops, looks like should change to `any[]` to be compatible

**Comment by Sorosliu1029** [comment]
Weird thing, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33190/commits/ba193d01ad698b99bc87f5e4402b74e348c9df35 fail on CI, but success on my own laptop. Help needed!

**Comment by amcasey** [comment]
@Sorosliu1029 sorry, we were having CI problems last night.  It looks like it's passing now.  Is this ready to go?

**Comment by Sorosliu1029** [comment]
yes

**Pull request merged**